packages/expo-cli/src/credentials/actions/list.ts

```diff
@@ -0,0 +1,55 @@
+import chalk from 'chalk';
+import uniq from 'lodash/uniq';
+import isEmpty from 'lodash/isEmpty';
+import fs from 'fs-extra';
+import get from 'lodash/get';
+import { AndroidCredentials as Credentials } from '@expo/xdl';
+import { AndroidCredentials } from '../credentials';
+
+import log from '../../log';
+
+export async function displayAndroidCredentials(credentialsList: AndroidCredentials[]) {
+  log(chalk.bold('Available Android credentials'));
+  log();
+  for(const credentials of credentialsList) {
+    await displayAndroidAppCredentials(credentials);
+  }
+}
+
+export async function displayAndroidAppCredentials(credentials: AndroidCredentials) {
+  const tmpFilename = `expo_tmp_keystore_file.jks`;
+  try {
+    if (await fs.pathExists(tmpFilename)) {
+      await fs.unlink(tmpFilename);
+    }
+
+    log(chalk.green(credentials.experienceName));
+    log(chalk.bold('  Upload Keystore hashes'));
+    if (!isEmpty(credentials.keystore)) {
+      const storeBuf = Buffer.from(get(credentials, 'keystore.keystore'), 'base64');
+      await fs.writeFile(tmpFilename, storeBuf);
+      await Credentials.logKeystoreHashes(
+        {
+          keystorePath: tmpFilename,
+          ...(credentials.keystore as Credentials.Keystore)
+        },
+        '    ',
+      );
+    } else {
+      log('    -----------------------');
+    }
+    log(chalk.bold('  Push Notifications credentials'));
+    log(
+      '    FCM Api Key: ',
+      get(credentials, 'pushCredentials.fcmApiKey', '---------------------')
+    );
+    log('\n');
+  } catch (error) {
+    log.error('  Failed to parse the keystore', error);
```

Comment:
I think this message won't be correct for every case ;)

Reply:
Only Keystore related errors can happen here. The only reason I added this catch is that otherwise, some mangled credentials in other apps could potentially stop the user from doing anything. Problem with the format of received that is the only issue I'm expecting here.